code/ecdsap256/Hacl.Impl.P256.LowLevel.fst

```diff
@@ -520,82 +595,288 @@ let mul1_add f1 u2 f3 result =
+    uint_v (Lib.Sequence.index (as_seq h4 out) 0) +  
+    uint_v (Lib.Sequence.index (as_seq h4 out) 1) * pow2 64 + 
+    uint_v (Lib.Sequence.index (as_seq h4 out) 2) * pow2 64 * pow2 64 + 
 
-  copy out temp; 
+    as_nat h4 b3 * pow2 64 * pow2 64 * pow2 64 + 
+    
+    uint_v (Lib.Sequence.index (as_seq h4 out) 7) * pow2 64 * pow2 64 * pow2 64 * pow2 64 * pow2 64 * pow2 64 * pow2 64;
+    
+    (==) {
+      lemma_mul0 (as_nat h4 b3) (uint_v (Lib.Sequence.index (as_seq h4 out) 7)) (as_nat h3 r) (uint_v f3) (as_nat h3 b3)
+     }
+
+    uint_v (Lib.Sequence.index (as_seq h4 out) 0) +  
+    uint_v (Lib.Sequence.index (as_seq h4 out) 1) * pow2 64 + 
+    uint_v (Lib.Sequence.index (as_seq h4 out) 2) * pow2 64 * pow2 64 + 
+    as_nat h3 r * uint_v f3 * pow2 64 * pow2 64 * pow2 64 + 
+    as_nat h3 b3 * pow2 64 * pow2 64 * pow2 64;
+
+    (==)
+    {
+      assert(Lib.Sequence.index (as_seq h3 bk2) 0 == Lib.Sequence.index (as_seq h3 out) 0);
+      assert(Lib.Sequence.index (as_seq h3 bk2) 1 == Lib.Sequence.index (as_seq h3 out) 1);
+      assert(Lib.Sequence.index (as_seq h3 bk2) 2 == Lib.Sequence.index (as_seq h3 out) 2)
+    }
+    
+    uint_v (Lib.Sequence.index (as_seq h3 out) 0) +  
+    uint_v (Lib.Sequence.index (as_seq h3 out) 1) * pow2 64 + 
+    uint_v (Lib.Sequence.index (as_seq h3 out) 2) * pow2 64 * pow2 64 + 
+    (
+      uint_v (Lib.Sequence.index (as_seq h3 out) 3) + 
+      uint_v (Lib.Sequence.index (as_seq h3 out) 4) * pow2 64 + 
+      uint_v (Lib.Sequence.index (as_seq h3 out) 5) * pow2 64 * pow2 64 + 
+      uint_v (Lib.Sequence.index (as_seq h3 out) 6) * pow2 64 * pow2 64 * pow2 64
+    ) * pow2 64 * pow2 64 * pow2 64 + 
+    as_nat h3 r * uint_v f3 * pow2 64 * pow2 64 * pow2 64;
   
-  pop_frame()
+    (==) 
+    {
+      lemma_mul1 (uint_v (Lib.Sequence.index (as_seq h3 out) 3)) 
+  (uint_v (Lib.Sequence.index (as_seq h3 out) 4)) (uint_v (Lib.Sequence.index (as_seq h3 out) 5)) (uint_v (Lib.Sequence.index (as_seq h3 out) 6))
+    }
+
+    uint_v (Lib.Sequence.index (as_seq h3 out) 0) +  
+    uint_v (Lib.Sequence.index (as_seq h3 out) 1) * pow2 64 + 
+    as_nat h3 b2 * pow2 64 * pow2 64 + 
+    uint_v (Lib.Sequence.index (as_seq h3 out) 6) * pow2 64 * pow2 64 * pow2 64 * pow2 64 * pow2 64 * pow2 64 + 
+    as_nat h3 r * uint_v f3 * pow2 64 * pow2 64 * pow2 64;
+    (==)
+    {
+      lemma_mul2 (as_nat h3 b2) (uint_v (Lib.Sequence.index (as_seq h3 out) 6)) (as_nat h2 r) (uint_v f2) (as_nat h2 b2)
+    }
+  
+    uint_v (Lib.Sequence.index (as_seq h3 out) 0) +  
+    uint_v (Lib.Sequence.index (as_seq h3 out) 1) * pow2 64 + 
+    
+    as_nat h2 r * uint_v f2 * pow2 64 * pow2 64 + 
+    as_nat h2 b2 * pow2 64 * pow2 64 + 
+    as_nat h3 r * uint_v f3 * pow2 64 * pow2 64 * pow2 64;
 
-#pop-options
+    (==)
+    {    
+      assert(Lib.Sequence.index (as_seq h2 bk1) 0 == Lib.Sequence.index (as_seq h2 out) 0);
+      assert(Lib.Sequence.index (as_seq h2 bk1) 1 == Lib.Sequence.index (as_seq h2 out) 1)
+    }
```

Comment:
why is `eq_intro` not working in those cases?

Reply:
Because I did not know about it's existence :)